### DISCUSSION TITLE
How to include custom headers in the request sent?

### DISCUSSION THREAD

**Discussion by IgnisDa** [oscartbeaumont/rspc#139] [rspc Help] [answered]
I am going through https://www.rspc.dev/client but I do not see a way to send custom headers in the request. For example, if I wish to send a `Bearer <token>` in the request.

**Comment by oscartbeaumont** [comment, marked as answer]
You can provide a custom fetch implementation [shown here](https://www.rspc.dev/client#custom-fetch-implementation) and extend the headers array.

So like the following:

```ts
import { createClient, FetchTransport } from "@rspc/client";
import type { Procedures } from "./bindings.ts"; // The bindings exported from your Rust code!
 
const client = createClient<Procedures>({
  transport: new FetchTransport(
    "http://localhost:4000/rspc",
    // Include Cookies for cross-origin requests
    (input, init) => fetch(input, { ...init, headers: { ...init.headers, "X-Demo": "123" })
  ),
});
```

In a future update, I will probs add a nicer way to do this because forgetting to pass through the `init.headers` will likely cause people some strife.
